src/main/resources/db/migration/V1_18__contract.sql

```diff
@@ -0,0 +1,34 @@
+create table contract (
+  id uuid not null,
+  type varchar(30) not null,
+
+  service_category_id uuid not null,
+  service_provider_id varchar(30) not null,
+
+  start_date timestamp with time zone not null,
+  end_date timestamp with time zone not null,
+
+  region_id char,
+  pcc_region_id text,
+
+  primary key (id),
+  constraint fk__contract__service_category_id foreign key (service_category_id) references service_category,
+  constraint fk__contract__service_provider_id foreign key (service_provider_id) references service_provider,
+  constraint fk__contract__pcc_region_id foreign key (pcc_region_id) references pcc_region,
+  constraint fk__contract__region_id foreign key (region_id) references region,
+  constraint contract_dates_valid check (start_date < end_date),
+  constraint contract_region_valid check
+      ((pcc_region_id is not null and region_id is null) or (pcc_region_id is null and region_id is not null))
+);
+
+create table contract_detail (
+    contract_id uuid not null,
+    age_group varchar(30) not null,
```

Comment:
could this be two integer fields, min_age and max_age, to make it clearer?

Reply:
Separated out